gn2/wqflask/templates/wiki/history.html

```diff
@@ -0,0 +1,141 @@
+    </style>
+{% endblock %}
+{% block content %}
+    {{ flash_me() }}
+    <div class = "container container-fluid">
+        <h2>GeneWiki Entry History</h2>
+        {% if most_recent %}
+            <h3>
+                <strong>Most Recent Version:</strong>
+            </h3>
+            <table class="table table-responsive table-bordered">
+                <tbody>
+                    <tr>
+                        <th>Gene Symbol:</th>
+                        <td>{{ most_recent["symbol"] }}</td>
+                    </tr>
+                    {% if most_recent["species"] %}
+                        <tr>
+                            <th>Species:</th>
+                            <td>{{ most_recent["species"] }}</td>
+                        </tr>
+                    {% endif %}
+                    {% if most_recent["pubmed_ids"] %}
+                        <tr>
+                            <th>PubMed IDs:</th>
+                            <td>
+                                {% for id in most_recent["pubmed_ids"] %}
+                                    <a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids={{ id }}&dopt=Abstract">{{ id }}</a>&nbsp;
+                                {% endfor %}
+                            </td>
+                        </tr>
+                    {% endif %}
+                    {% if most_recent["web_url"] %}
+                        <tr>
+                            <th>Web URL:</th>
+                            <td>{{ most_recent["web_url"] }}</td>
+                        </tr>
+                    {% endif %}
+                    <tr>
+                        <th>Entry:</th>
+                        <td>{{ most_recent["comment"] }}</td>
+                    </tr>
+                    {% if most_recent["categories"] %}
+                        <tr>
+                            <th>Category:</th>
+                            <td>{{ '; '.join(most_recent["categories"]) }}</td>
+                        </tr>
+                    {% endif %}
+                    <tr>
+                        <th>Add Time:</th>
+                        <td>{{ most_recent["created"] }}</td>
+                    </tr>
+                    {% if most_recent["reason"] %}
+                        <tr>
+                            <th>Reason for Modification:</th>
+                            <td>{{ most_recent["reason"] }}</td>
+                        </tr>
+                    {% endif %}
+                </tbody>
+            </table>
+        {% endif %}
+        <h3>
+            <strong>Previous Version:</strong>
+        </h3>
+        {% if previous_versions %}
+            {% for version in previous_versions %}
+                <table class="table table-responsive table-bordered">
+                    <tr>
+                        <th>Gene Symbol:</th>
+                        <td>{{ version["symbol"] }}</td>
+                    </tr>
+                    {% if version["species"] %}
```

Comment:
I think it would be better to display the rows even if they didn't exist and provide a sane sentinel value e.g. `{{ version.get("species", "not available") }}` which serves two functions:

- makes the template easier to read and parse since we drop all the ifs
- shows the user that this value doesn't exists.

This also applies to mos of the other ifs